tests/test_client.py

```diff
@@ -0,0 +1,81 @@
+import sys
+from unittest.mock import MagicMock, patch
+
+import pytest
+from httpx import HTTPStatusError
+from pkg_resources import DistributionNotFound
+
+from dolib import AsyncClient, Client
+from dolib.client import BaseClient
+
+
+@patch("pkg_resources.get_distribution", side_effect=DistributionNotFound)
+def test_version(mock: MagicMock) -> None:
+    del sys.modules["dolib.__version__"]
+    from dolib.__version__ import __version__
+
+    assert __version__ is None
+
+
+def test_base_client() -> None:
+    with pytest.raises(ValueError):
```

Comment:
```suggestion
    with pytest.raises(ValueError, match=r"^Regexp string$"):
```

`ValueError` is too broad. Matching the message sting could help.